portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,55 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.gson.Gson;
 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import java.util.ArrayList;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<Comment> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Juan!</h1>");
+    response.setContentType("application/json;");
+    String json = getCommentsJson();
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Comment newComment = getComment(request);
+    comments.add(newComment);
+    response.sendRedirect("/index.html");
+  }
+
+  public Comment getComment(HttpServletRequest request) {
+    String firstName = request.getParameter("inputFirstName");
+    String lastName = request.getParameter("inputLastName");
+    String message = request.getParameter("inputComment");
+    return new Comment(firstName,lastName,message);
+  }
+  
+  public String getCommentsJson() {
```

Comment:
Just for something to think about (you don't need to change anything) how would you test this function? It seems like it would be pretty tricky since comments are added in the `doPost` method. I think you would have to create a bunch of fake HTTP requests to process. If this function accepted a `List<Comment>` instead, then you could just create comments in your test and pass them in. 

This kind of stuff falls under the category of `dependency injection` which is a ridiculously over complicated way of saying "pass in the stuff this thing is going to need". I find in practice it doesn't make any difference other than that writing tests is much easier.

Reply:
That's a good point I hadn't thought about. I had to change the method anyway to take a parameter because I added datastore (new pull request coming soon).